head/src/main/kotlin/io/qalipsis/core/head/hook/ValidationCampaignHook.kt

```diff
@@ -98,12 +98,12 @@ internal class ValidationCampaignHook(
         }
     }
 
-    private fun validateZones(zones: Map<String, Int>, constraintsViolations: MutableCollection<String>) {
+    private suspend fun validateZones(tenant: String, zones: Map<String, Int>, constraintsViolations: MutableCollection<String>) {
 
         /**
          * Keys of the zones supported by the cluster.
          */
-        val allowedZones = headConfiguration.cluster.zones.map { it.key }.toSet()
+        val allowedZones = zoneService.list(tenant).map { it.key }.toSet()
```

Comment:
Only fetch the zones when `zones` is not empty, otherwise you so this expensive operation for nothing.